paddle/gserver/layers/ContextProjection.h

```diff
@@ -41,7 +41,7 @@ namespace paddle {
  *
  * The config file api is context_projection.
  */
-class ContextProjection : public Projection {
+class ContextProjection : public Projection, public WithFunction {
```

Comment:
There have been many discussions about why we need to avoid C++ multiple inheritances.  E.g., http://stackoverflow.com/a/407928/724872.  Instead, we should use "composition".